src/service-worker.js

```diff
@@ -169,8 +171,14 @@ self.addEventListener('fetch', event => {
 self.addEventListener('push', event => {
   event.waitUntil((async () => {
     const data = event.data.json()
-    const { origin } = event.target
+    const knownInstances = await getKnownInstances()
+    if (knownInstances.length !== 1) {
+      // TODO: find a way to determine the pushing instance from the PushEvent
+      await showSimpleNotification(data)
```

Comment:
It looks like we are eagerly showing a simple notification rather than a rich one, whereas the original logic was to show a rich one and fallback to a simple one. Why would we prefer to always show a simple notification rather than a rich one, when there is only one instance in the known instances list?

Reply:
More details in https://github.com/nolanlawson/pinafore/issues/1663#issuecomment-1336548657

The problem is that there's no way to compute the `origin` from the `event` alone.

Reply:
The idea here is that:
* if we only know about a single instance, the notification should come from it so we use it (line 181) and show a rich notification
* otherwise, we have no way to determine the `origin`, so fallback to a simple notification.

Pending deeper changes on the way enabling notifications is handled (e.g. only allow enabling notifications on a single instance) or changes on Mastodon's side (including the URL to the notification, or at least the instance, in the PushEvent), this PR only fixes things in the case where the user logged in to a single instance. Not ideal but that's the simplest change to at least get rich notifications for some users (most of them? a good proportion of them? I'm certainly one of them).